maker/src/routes.rs

```diff
@@ -140,6 +141,46 @@ pub async fn post_sell_order(
     Ok(())
 }
 
+/// The maker POSTs this to set the offer params
+#[derive(Debug, Clone, Deserialize)]
+pub struct CfdNewOfferParamsRequest {
+    pub price_long: Option<Price>,
+    pub price_short: Option<Price>,
+    pub min_quantity: Usd,
+    pub max_quantity: Usd,
+    pub tx_fee_rate: Option<TxFeeRate>,
+    pub funding_rate: Option<FundingRate>,
+    // TODO: This is not inline with other parts of the API! We should not expose internal types
+    // here. We have to specify sats for here because of that.
+    pub opening_fee: Option<OpeningFee>,
+}
+
+#[rocket::post("/offer", data = "<offer_params>")]
```

Comment:
I think I would have a mild preference of `PUT` over `POST`.

Conventionally, `PUT` is a _replace_ action. `PUT` is also required to be idempotent which this operation is!

Reply:
I'll give it a shot, thanks for reminding about it!